core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/aggregator/Aggregator.java

```diff
@@ -0,0 +1,295 @@
+        for(int i=0; i<aggregationItems.size(); i++)
+        {
+            Schema.checkAttributeExists(inputSchema, aggregationItems.get(i).getAttributeName());
+            Schema.checkAttributeNotExists(inputSchema, aggregationItems.get(i).getResultAttributeName());
+        }
+
+        for(int i=0; i<aggregationItems.size(); i++)
+        {
+            if(aggregationItems.get(i).getAggregatorType() != AggregationType.COUNT)
+            {
+                if(Schema.isAttributeType(inputSchema, aggregationItems.get(i).getAttributeName(), AttributeType._ID_TYPE)
+                   || Schema.isAttributeType(inputSchema, aggregationItems.get(i).getAttributeName(), AttributeType.DATE)
+                   || Schema.isAttributeType(inputSchema, aggregationItems.get(i).getAttributeName(), AttributeType.LIST)
+                   || Schema.isAttributeType(inputSchema, aggregationItems.get(i).getAttributeName(), AttributeType.STRING)
+                   || Schema.isAttributeType(inputSchema, aggregationItems.get(i).getAttributeName(), AttributeType.TEXT))
+                {
+                    throw new TexeraException(
+                            ErrorMessages.ATTRIBUTE_TYPE_NOT_FIT_FOR_AGGREGATION(aggregationItems.get(i).getAttributeName(),  aggregationItems.get(i).getAggregatorType().toString()));
+                }
+            }
+        }
+
+        Builder schemaBuilder = new Schema.Builder();
+        for(int i=0; i<aggregationItems.size(); i++)
+        {
+            schemaBuilder = schemaBuilder.add(aggregationItems.get(i).getResultAttributeName(), AttributeType.DOUBLE);
+        }
+        
+        outputSchema = schemaBuilder.build();
+    }
+
+    @Override
+    protected Tuple computeNextMatchingTuple() throws TexeraException
+    {
+        if (cursor == CLOSED || cursor >= 1) {
+            return null;
+        }
+        Tuple resultTuple = null;
+        resultTuple = processAllTuples(inputOperator);
+        cursor++;
+        return resultTuple;
+    }
+
+    public Tuple processAllTuples(IOperator inputOperator)
+    {
+        Tuple inputTuple;
+        List<IField> aggregatedResults = initialiseResultFieldsList();
+        
+        List<AggregationAttributeAndResult> aggregationItems = predicate.getAttributeAggregateResultList();
+        
+        while ((inputTuple = inputOperator.getNextTuple()) != null)
+        {
+            //incrementing the count
+            aggregatedResults.set(0, new IntegerField((int)aggregatedResults.get(0).getValue() + 1));
+            
+            for(int i=0; i<aggregationItems.size(); i++)
+            {
+                IField field = inputTuple.getField(aggregationItems.get(i).getAttributeName());
+                Attribute inputAttr= inputSchema.getAttribute(aggregationItems.get(i).getAttributeName());
+                
+                switch(aggregationItems.get(i).getAggregatorType())
```

Comment:
These switch statements have lots of things in common. Is it possible to extract the common things and simplify them? You can possibly create a generic function to do the comparison similar to this:
`private <T> int compare(T a, Tb)`

Reply:
This comment hasn't been addressed yet. 

@avinash0161 any thoughts on implementing a generic compare function to avoid duplicating the same piece of code for different data types?

Furthermore, a more important problem being the code only handles `integer` and `double` data type. Other data types, such as `date`, `string`, and `text` should also be supported for `min` and `max`. (although `sum` and `avg` doesn't make sense for them)

Reply:
Date, String and text are already been considered for max and min.

I changed the repeated code to use compare(T a, T b). However, the number of lines of code haven't changed because initially also the decision was made in one line of code. Do you have any idea as to how to decrease the code further?